store/mongo/datastore_mongo.go

```diff
@@ -417,9 +440,36 @@ func (db *DataStoreMongo) DeleteTokenByDevId(
 	ctx context.Context,
 	devID oid.ObjectID,
 ) error {
-	c := db.client.Database(ctxstore.DbFromContext(ctx, DbName)).
+	c := db.client.Database(DbName).
 		Collection(DbTokensColl)
-	ci, err := c.DeleteMany(ctx, bson.M{"sub": devID})
+	var filter interface{}
+	id := identity.FromContext(ctx)
+	if id != nil {
+		// in here if mender.tenant is not set we will not match,
+		// since there is no way to tell looking at id if we wanted empty Tenant
+		// or nil. we could have either change AddToken or remove omitempty.
+		if id.Tenant == "" {
+			filter = bson.M{
+				"$or": []bson.M{
+					{"sub": devID},
+					{
+						"$and": []bson.M{
+							{"sub": devID},
+							{dbFieldTenantClaim: ""},
+						},
+					},
+				},
+			}
+		} else {
+			filter = bson.D{
+				{Key: "sub", Value: devID},
+				{Key: dbFieldTenantClaim, Value: id.Tenant},
+			}
+		}
+	} else {
+		filter = bson.M{"sub": devID}
+	}
```

Comment:
The `mender.tenant` claim is either present and non-empty or not present. An empty string would be a bug.
```suggestion
	filter := bson.M{"sub": devID, dbFieldTenantClaim: nil}
	id := identity.FromContext(ctx)
	if id != nil && id.Tenant != "" {
		filter[dbFieldTenantClaim] = id.Tenant
	}
```

Reply:
thank you lovely. wow, null, who would have thought, right? :)